### PR TITLE
Set up pre-commit to generate github yaml from nickel

### PR DESCRIPTION
**Pull request by vkleen**
This is a bit of an unpolished experiment. But if generating the YAML for github actions with Nickel works, that would be awesome 🎉

### CODE REVIEW

File: .github/settings.yml [hunk 1]
```diff
@@ -1,40 +1,74 @@
...
+    {
+      "color": "0e8a16",
+      "description": "merge on green CI",
+      "name": "merge-queue"
+    }
+  ],
+  "repository": {
+    "has_wiki": false
+  }
+}
```
Comment (aspiwack) [inline .github/settings.yml:74]: Maybe make sure that there's a newline at the end of generated files as Git likes it better.

File: flake.nix [hunk 2]
```diff
@@ -81,6 +81,14 @@
             vendorHash = "sha256-HQQQlOqjI4EAB1JNhFmjt9SQrF2wOh6C7hrR8So0dxs=";
           };
 
+          eval-github-yaml = pkgs.writeShellScript "eval-github-yaml" ''
+            set -e
+            for f in "$@"; do
+              (echo "# DO NOT EDIT! Generated by eval-github-yaml from $f"; \
+                ${inputs.nickel.packages.${system}.default}/bin/nickel export --format json -f "$f") > "''${f%.ncl}.yml"
```
[the remark this answers is not in the document]
Comment (vkleen) [reply flake.nix:88]: I actually did that at first, but dependabot chokes on "00:00" being interpreted as a sexagesimal number when it's not quoted. That's probably something we should address in Nickel, but for now exporting as JSON makes sure that something like that doesn't happen.

[the remark this answers is not in the document]
Comment (aspiwack) [reply flake.nix:88]: I guess this is an upstream bug in the Yaml exporter that we use. Cc @yannham .

[the remark this answers is not in the document]
Comment (yannham) [reply flake.nix:88]: Right, looks like so.

[the remark this answers is not in the document]
Comment (yannham) [reply flake.nix:88]: @vkleen could you please open an issue against the Nickel repo?

File: .github/settings.yml (hunk 1 above)
Comment (vkleen) [reply .github/settings.yml]: Very good point; addressed by patching Nickel proper 👍

File: flake.nix (hunk 2 above)
[the remark this answers is not in the document]
Comment (vkleen) [reply flake.nix:88]: https://github.com/tweag/nickel/issues/1222

**Pull request merged**